packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -34,20 +34,25 @@ class UrlLauncherPlugin extends UrlLauncherPlatform {
     UrlLauncherPlatform.instance = UrlLauncherPlugin();
   }
 
-  String _getUrlScheme(String url) => Uri.tryParse(url)?.scheme;
+  String _getUrlScheme(String url) =>
+      Uri
+          .tryParse(url)
+          ?.scheme;
 
   bool _isSafariTargetTopScheme(String url) =>
       _safariTargetTopSchemes.contains(_getUrlScheme(url));
 
-  /// Opens the given [url] in a new window.
+  /// Opens the given [url] in the specified [webOnlyLinkTarget].
   ///
   /// Returns the newly created window.
   @visibleForTesting
-  html.WindowBase openNewWindow(String url) {
+  html.WindowBase openNewWindow(String url,{webOnlyLinkTarget} ) {
     // We need to open mailto, tel and sms urls on the _top window context on safari browsers.
     // See https://github.com/flutter/flutter/issues/51461 for reference.
     final target =
-        browser.isSafari && _isSafariTargetTopScheme(url) ? '_top' : '';
+    browser.isSafari && _isSafariTargetTopScheme(url)
+        ? '_top'
+        : webOnlyLinkTarget ?? "";
```

Comment:
Reading the [original ticket](https://github.com/flutter/flutter/issues/56867#issuecomment-629363114), I think this logic is backwards:

* If the user is passing a webOnlyLinkTarget (!= null), use that
* Else, use the older logic

So I guess this needs to be:

```dart
final target = webOnlyLinkTarget ?? browser.isSafari && _isSafariTargetTopScheme(url) ? '_top' : '';
```